plugins/jobs/listBuilds.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+const boom = require('boom');
+const hoek = require('hoek');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const listSchema = joi.array().items(schema.models.build.get).label('List of builds');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/jobs/{id}/builds',
+    config: {
+        description: 'Get builds for a given job',
+        notes: 'Returns builds for a given job',
+        tags: ['api', 'jobs', 'builds'],
+        handler: (request, reply) => {
+            const factory = request.server.app.jobFactory;
+
+            return factory.get(request.params.id)
+              .then(job => {
+                  if (!job) {
+                      throw boom.notFound('Job does not exist');
+                  }
+
+                  let config = {};
+
+                  if (Object.keys(request.query).length !== 0) {
```

Comment:
We should never be doing things like this, instead we should be using the default feature of Joi.